test/tstInterpDetailsSVD.cpp

```diff
@@ -0,0 +1,216 @@
+  using MemorySpace = typename DeviceType::memory_space;
+  using view_t = Kokkos::View<double **, MemorySpace>;
+  ExecutionSpace space{};
+
+  double mat0[2][2] = {{1, 2}, {2, 3}};
+  double inv0[2][2] = {{-3, 2}, {2, -1}};
+  makeCase<ExecutionSpace, view_t, double[2][2]>(space, 0, mat0, inv0, 2);
+
+  double mat1[2][2] = {{4, 0}, {0, 4}};
+  double inv1[2][2] = {{1 / 4., 0}, {0, 1 / 4.}};
+  makeCase<ExecutionSpace, view_t, double[2][2]>(space, 1, mat1, inv1, 2);
+
+  double mat2[2][2] = {{0, 5}, {5, 0}};
+  double inv2[2][2] = {{0, 1 / 5.}, {1 / 5., 0}};
+  makeCase<ExecutionSpace, view_t, double[2][2]>(space, 2, mat2, inv2, 2);
+
+  double mat3[2][2] = {{2, 2}, {2, 2}};
+  double inv3[2][2] = {{1 / 8., 1 / 8.}, {1 / 8., 1 / 8.}};
+  makeCase<ExecutionSpace, view_t, double[2][2]>(space, 3, mat3, inv3, 2);
+
+  double mat4[2][2] = {{1, -3}, {-3, 1}};
+  double inv4[2][2] = {{-1 / 8., -3 / 8.}, {-3 / 8., -1 / 8.}};
+  makeCase<ExecutionSpace, view_t, double[2][2]>(space, 4, mat4, inv4, 2);
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(pseudo_inv_symm3, DeviceType, ARBORX_DEVICE_TYPES)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+  using MemorySpace = typename DeviceType::memory_space;
+  using view_t = Kokkos::View<double **, MemorySpace>;
+  ExecutionSpace space{};
+
+  double mat0[3][3] = {{2, 2, 3}, {2, 0, 1}, {3, 1, -2}};
+  double inv0[3][3] = {{-1 / 18., 7 / 18., 2 / 18.},
+                       {7 / 18., -13 / 18., 4 / 18.},
+                       {2 / 18., 4 / 18., -4 / 18.}};
+  makeCase<ExecutionSpace, view_t, double[3][3]>(space, 0, mat0, inv0, 3);
+
+  double mat1[3][3] = {{0, 1, 2}, {1, 2, 3}, {2, 3, 4}};
+  double inv1[3][3] = {{-5 / 6., -1 / 6., 3 / 6.},
+                       {-1 / 6., 0, 1 / 6.},
+                       {3 / 6., 1 / 6., -1 / 6.}};
+  makeCase<ExecutionSpace, view_t, double[3][3]>(space, 1, mat1, inv1, 3);
+
+  double mat2[3][3] = {{5, 5, 5}, {5, 5, 5}, {5, 5, 5}};
+  double inv2[3][3] = {{1 / 45., 1 / 45., 1 / 45.},
+                       {1 / 45., 1 / 45., 1 / 45.},
+                       {1 / 45., 1 / 45., 1 / 45.}};
+  makeCase<ExecutionSpace, view_t, double[3][3]>(space, 2, mat2, inv2, 3);
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(pseudo_inv_symm128, DeviceType,
+                              ARBORX_DEVICE_TYPES)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+  using MemorySpace = typename DeviceType::memory_space;
+  using view_t = Kokkos::View<double **, MemorySpace>;
+  ExecutionSpace space{};
+
+  // 128x128 matrix full of -2
+  // eigenvalues are -2*128 and 127 0s
+  // pseudo-inverse is a 128x128 matrix of 1 / (128 * 128 * -2)
+  double mat[128][128] = {};
+  double inv[128][128] = {};
+  for (int i = 0; i < 128; i++)
+    for (int j = 0; j < 128; j++)
+    {
+      mat[i][j] = -2;
+      inv[i][j] = 1 / (128 * 128 * -2.);
+    }
+  makeCase<ExecutionSpace, view_t, double[128][128]>(space, 0, mat, inv, 128);
+
+  // Case for invertible 128x128 matrix
```

Comment:
If I do find an way to easily build a non-trivial 128x128 matrix and its inverse. Since I added this comment I did not found one so I will remove that command. This test could be extended if I ever find said matrix.